Added a test class for Basic64 class, all tests disabled except one. …

agdturner opened the pull request:
…Copyright assigned to Ben Evans. MIT license applied to both Basic64.java and Basic64Test.java

kittylyst reviewed (changes requested):
Hi @agdturner - sorry it's taken me a while to get to this. I think the core of this is sound, but there are a couple of problems:

1. Lots of autogenerated boilerplate test code that is all @Disabled - I don't really see the value of checking this in
2. Changes in formatting that obscure the actual content of the change
3. Addition of license files  

I was deliberately holding off on 3. until there was a better sense of which license to choose. It will *probably* be either Apache or MIT - but I wanted to think on it some more. Can you remove the license files for now?

As to 2. - I will undertake to set up automatic code-formatting as part of the build. For now, can you update this PR so that minor reformats do not appear as part of it?

Otherwise, looks pretty good - thank you!

kittylyst commented:
I've implemented automatic formatting (https://github.com/kittylyst/jDEC64/pull/23) and reformatted the codebase into the new style. Can you rebase this PR on the new style?

agdturner commented:
Hi Ben,

Good to hear from you.

It has been a while since I did this and I might not remember all
the details, I can check and should, but I don't have time for at least
the next 2 weeks...

I'm not sure how to do what you ask and I'm a bit confused. I looked into
editing a pull request and think the advice is to just make further changes
and try again. Is that what you think should happen?

What is a good way to do this? I was thinking that maybe I try to update my
fork from main and deal with any merge conflicts... I think I only changed
one file and created a test file for that class, so I can store my copy of
these files and perhaps just copy and paste the bits in that are wanted and
then let you know?

I use Netbeans as an IDE. When I create a JUnit test class (or update the
existing test class) Netbeans adds all method test boilerplate by default.
There are ways to configure what is added, but I've always just gone with
the default... As I understand it, that is a good way forward generally.
Next step is that I disable the tests that have not been written yet by
adding the @Disabled annotation. Then delete these @Disabled annotations as
tests are implemented or if not necessary be explicit about whch tests
cover the method using {@link #test...} in the comments. This gradually
leads to a more complete test suite and saves breaking the build per se. I
guess that is not normal in your experience. I can remove the boilerplate,
but I don't see any benefit in doing so. The advantage of it is that it
provides a clear track for others that want to collaborate and gives me
something to work on...

For 2 do you mean that it doesn't matter how the formatting of white
space/comments is for each of us and we can somehow apply a style to keep
things consistent in main but work with the style we are used to locally? I
want to work in the style I am used to, but to apply a style as needed to
contribute to main without all the changes just looking like white
space/comment format changes. Is there an easy way that I can work with my
preferred style in Netbeans and apply a different style before generating a
pull request? Maybe that is what you are suggesting with the build tool? In
the past I have found that auto-formatting styles work well for everything
but blank lines...

As I understand things, assigning you the copyright gives you liberty to
change the licence, but please let us just assign any licence for now
that allows you to relicense. IANAL and only have a basic understanding of
copyright and IP law. It is complicated as I think we are acting as
individuals in different jurisdictions, so our collaboration has a confused
legal basis. Often there is an umbrella organisation based in a particular
jurisdiction that we can use as a legal basis for our own safety in
collaboration. This is not part of work done under the Oracle/OpenJDK
contributor agreement as I understand it yet. I expect Oracle have some
advice on licensing... I just added a header in the source class files
stipulating a licence rather than adding a LICENCE file for the entire code
base...

Sorry, I am going from memory and need to check. I'll try to find time to
look at this again in a couple of weeks. Please send any feedback when you
can. I'm confident once we get the workflow right and smooth that
development can be rapid time permitting. For me, this is a busy time of
year generally and in the UK we have some big problems with pandemic and I
am having to cover lots of work for colleagues that I wouldn't normally
have to. I'm sure you understand, but as a volunteer project, this is way
down i the priority list of things I have to do.

Best wishes,

Andy

On Sat, Feb 6, 2021 at 10:27 AM Ben Evans <notifications@github.com> wrote:

> I've implemented automatic formatting (#23
> <https://github.com/kittylyst/jDEC64/pull/23>) and reformatted the
> codebase into the new style. Can you rebase this PR on the new style?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kittylyst/jDEC64/pull/22#issuecomment-774443047>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABM3GQSTPBGULYFM5OWYV73S5UKRRANCNFSM4V5XA7BQ>
> .
>
